base9/Settings.cpp

```diff
@@ -509,6 +511,24 @@ template<typename T> T Settings::getOrDie (Node & n, string && f)
     }
 }
 
+template<typename T> T Settings::getOrRequest (Node & n, string && f)
+{
+    if (n[f])
+    {
+        return n[f].as<T> ();
+    }
+    else
+    {
+        string input = "";
+        cout << "Please enter your desired setting for field: '" + f + "':\n";
+        getline(cin, input);
+        cout << "Field '" + f + "' set to: " << input << endl;
+        YAML::Node node;
```

Comment:
We already pull in `std::istringstream`, so why not

```cpp
    getline(cin, input)

    T t;
    istringstream(input) >> t;

    return t;
```

to get directly to the answer?